### ISSUE TITLE
iOS 11, Swift 4, Xcode 9

### ISSUE THREAD

**Issue by HackShitUp** [exyte/ReadabilityKit#24]
Just a heads up, some parsing in Readability doesn't work for iOS 11 and Swift 4. If you'd like, I can take a look into these issues and commit some changes to another branch?

**Comment by zapletnev** [comment]
@HackShitUp thanks for reporting! It would be great if you have a chance to check it out. Feel free to submit PR.

**Comment by HackShitUp** [comment]
Hi @zapletnev 
So I just checked the error, and it persistently logs the following:

XPath error : Invalid predicate
XPath error : Invalid expression

This error wasn't in Readability itself, but in the framework, "Ji." I'm not too familiar with parsing XML and HTML. Where and how are you using Ji to parse image data in Readability?

**Comment by zapletnev** [comment]
Hi @HackShitUp!
We will install Xcode beta and check it our asap. I will keep you informed.

**Comment by HackShitUp** [comment]
Awesome, thank you!

**Comment by honghaoz** [comment]
The related issue: https://github.com/honghaoz/Ji/issues/45

**Comment by HackShitUp** [comment]
@honghaoz beat me to it 😅

**Comment by mann579** [comment]
Is it resolved? As the errors are still there and readability returning null in success block.

**Comment by zapletnev** [comment]
Hi @mann579,
We are working on swift 4 release. It should be available on this or next week.

**Comment by vhailor13** [comment]
Just released 0.7.0 version (swift 4 support). It is available via:
pod 'ReadabilityKit', '0.7.0'

**Comment by HackShitUp** [comment]
@vhailor13 I just updated via Cocoapods, and it logs the following 2 errors:
``` swift
.../Pods/ReadabilityKit/Dependencies/libxml2/module.modulemap:2:14: Umbrella directory '/usr/include/libxml2' not found

Could not build Objective-C module 'ReadabilityKit'
```

**Comment by HackShitUp** [comment]
Perhaps something's wrong with the architecture?

**Comment by vhailor13** [comment]
Try to reinstall entire workspace: remove _Podfile.lock_, _Pods/_,  _*.xcworkspace_. And then run **pod install**

**Comment by atbrennan** [comment]
@HackShitUp @vhailor13 i got the libxml2 error as well, and removing podfile.lock / pods / xcworkspace has not fixed the issue for me. has that definitely fixed the issue for others, or is there another fix?

**Comment by JustinBobo** [comment]
I'm also experiencing the same issue at @atbrennan

**Comment by p3scobar** [comment]
Same.

**Comment by zapletnev** [comment]
Thanks for reporting @p3scobar @p3scobar @atbrennan @HackShitUp 
We released new version 0.7.1 that includes the fix. Please check it.

```
use_frameworks!

target 'SomeTarget' do
  pod 'ReadabilityKit', '0.7.1'
end
```
